cmd/nephe-controller/options.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v2"
+
+	"antrea.io/nephe/pkg/controllers/config"
+	"antrea.io/nephe/pkg/logging"
+)
+
+type Options struct {
+	// The path of configuration file.
+	configFile string
+	// The configuration object
+	config *config.ControllerConfig
+	Log    logr.Logger
+}
+
+func newOptions() *Options {
+	return &Options{
+		config: &config.ControllerConfig{},
+		Log:    logging.GetLogger("controllers").WithName("Options"),
+	}
+}
+
+func (o *Options) addFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.configFile, "config", o.configFile, "The path to the configuration file")
+}
+
+// complete completes all the required options.
+func (o *Options) complete() error {
+	if len(o.configFile) > 0 {
+		if err := o.loadConfigFromFile(); err != nil {
+			return fmt.Errorf("error loading config file %v, err: %v", o.configFile, err)
+		}
+	}
+	err := o.validate()
+	if err != nil {
+		return err
+	}
+	o.setDefaults()
+	return nil
+}
+
+func (o *Options) loadConfigFromFile() error {
+	data, err := os.ReadFile(o.configFile)
+	if err != nil {
+		return err
+	}
+	return yaml.UnmarshalStrict(data, &o.config)
+}
+
+func (o *Options) validate() error {
+	if len(o.config.CloudResourcePrefix) > 0 {
+		reg := regexp.MustCompile(`^[a-zA-Z0-9]+([_ -]?[a-zA-Z0-9])*$`)
+		valid := reg.MatchString(o.config.CloudResourcePrefix)
```

Comment:
Should we allow both _ and - in the prefix naming convention?

Reply:
Solved